### PR TITLE
Feature/cri 663 Offline testing

### PR DESCRIPTION
**Pull request by JakubJankowski**
jira: https://nomtek.atlassian.net/browse/CRI-663

Notes: Please use the new TestConfiguration.swift file, newest version is pinned to slack's ctb-ios channel. In that file, the new `mode` enum dictates the api's working mode - `useAPI` makes the tests run the old way, `useAPIAndRecord` makes the api run the tests the old way and record responses to files in device's documents directory, and `useRecordedResponses` causes the api to read responses from recorded files.

Also please note that `suffix` String in `TestConfiguration.swift` has to be manually changed every time tests are run in the `useAPIAndRecord` mode to make sure names are unique (for example when sending a create new group request).

`APIClientSpec` is unaffected by changes in this pull request, and still makes api calls.

### CODE REVIEW

File: CreatubblesAPIClient/Sources/Core/Error/APIClientError.swift [hunk 1]
```diff
@@ -80,7 +81,31 @@ open class APIClientError: Error
...
+        aCoder.encode(code, forKey: "code")
+        aCoder.encode(title, forKey: "title")
+        aCoder.encode(source, forKey: "source")
+        aCoder.encode(detail, forKey: "detail")
+        aCoder.encode(domain, forKey: "domain")
     }
+    
+    public required init?(coder aDecoder: NSCoder)
+    {
+        self.status = aDecoder.decodeInteger(forKey: "status")
```
Comment (mmiedlarz) [inline CreatubblesAPIClient/Sources/Core/Error/APIClientError.swift:101]: I think, that it's better to use `!` operators here, and keep properties non-optional.

File: CreatubblesAPIClientTests/TestUtils/Offline Testing/RecorderResponseHandler.swift [hunk 2]
```diff
@@ -0,0 +1,45 @@
+//
+//  RecorderResponseHandler.swift
+//  CreatubblesAPIClient
+//
+//  Created by Nomtek on 28.11.2016.
```
Comment (mmiedlarz) [inline CreatubblesAPIClientTests/TestUtils/Offline Testing/RecorderResponseHandler.swift]: Please use proper (MIT) license header

File: CreatubblesAPIClientTests/TestUtils/Offline Testing/RecorderTestSender.swift [hunk 3]
```diff
@@ -0,0 +1,141 @@
+//
+//  RecorderTestSender.swift
+//  CreatubblesAPIClient
+//
+//  Created by Nomtek on 28.11.2016.
+//  Copyright © 2016 Nomtek. All rights reserved.
+//
+
+import UIKit
+@testable import CreatubblesAPIClient
```
Comment (mmiedlarz) [inline CreatubblesAPIClientTests/TestUtils/Offline Testing/RecorderTestSender.swift:27]: Discussed via Slack

**Review by mmiedlarz — CHANGES_REQUESTED** [review]

**Comment by DaDer** [comment]
Code: +1
Func: +1 (sometimes on recording some tests are failing, but this issue is not related to the task)

**Pull request merged**